tests/Mapper/ORM/DoctrineORMMapperTest.php

```diff
@@ -62,6 +65,39 @@ public function testDoctrineMappingLoaded(): void
         static::assertInstanceOf(DoctrineORMMapper::class, $mapper);
     }
 
+    public function testLoadClassMetadata(): void
+    {
+        self::bootKernel();
+
+        /** @var DoctrineORMMapper $mapper */
+        $mapper = static::getContainer()->get('sonata.doctrine.mapper');
+
+        /** @var EntityManagerInterface $entityManager */
+        $entityManager = static::getContainer()->get('doctrine.orm.entity_manager');
+
+        $classMetadata = $entityManager->getClassMetadata(TestEntity::class);
+        $classMetadata->setDiscriminatorColumn([
+            'name' => 'discriminator',
+            'length' => 10,
+        ]);
+
+        $mapper->loadClassMetadata(new LoadClassMetadataEventArgs($classMetadata, $entityManager));
+
+        $expectedDefinition = [
+            'type' => 'string',
+            'fieldName' => 'discriminator',
+            'name' => 'discriminator',
+            'length' => 10,
+        ];
+
+        static::assertEquals(
```

Comment:
hm... we don't allow `assertEquals`?

https://github.com/sonata-project/sonata-doctrine-extensions/actions/runs/7790592026/job/21244791927?pr=494

why? I mean its a perfectly valid use case for it :confused: I want to loosely compare the objects or arrays. Whats the workaround for allowing this then?

Reply:
I think assertEquals is too loose, for instance.
```
class Foo {
    public function construct(private $id) {};
}

static::assertEquals(new Foo (42), new Foo('42')); // true
static::assertEquals(new Foo (0), new Foo(false')); // true
```

Maybe we could assertSame((array) $a, (array) $b) ?

Reply:
The order differs. Will try to sort the array keys then I guess